### PR TITLE
#151709633 Notification when contribution is deleted.

### PR DESCRIPTION
**Pull request by iankigen**
#### What does this PR do?

- Add stream notification on contribution deletion to the contributor

#### Description of Task to be completed?

- Update contribution delete endpoint to create notification for contributor
- Create a channel to broadcast new notification for contributor
- Write tests for the functionality

#### How should this be manually tested?

``` pytest -v ```

#### What are the relevant pivotal tracker stories?

#151709633

#### Screenshots

capturfiles_12
capturfiles_13

### CODE REVIEW

File: api/views/contributions.py [hunk 1]
```diff
@@ -288,6 +288,21 @@ def delete(self, contrib_id):
...
+                      f'has been deleted.'
+            contibutor_notification = Notification.create(
+                message,
+                'contribution',
+                _contribution.id,
+                _contribution.user_id,
+                userId=current_user.user_id,
+                userName=current_user.name,
+                userImg=current_user.image_url)
+            sse.publish({"data": contibutor_notification.save(),
```
Comment (georgreen) [inline api/views/contributions.py]: Can this block publish block be moved, to occur after the delete has been executed?

File: test/api/views/test_notifications.py [hunk 2]
```diff
@@ -179,3 +214,18 @@ def test_user_patch_all_notification_read_status_to_True(self):
...
+            content_type='application/json', headers=self.admin_header)
+        user_notifications_after = Notification.get_count(self.regular_user.id)
+        self.assertTrue(user_notifications_after > user_notifications_before)
+
+    @patch('flask_sse.sse.publish')
+    def test_delete_contribution_notification_stream(self, mock):
+        response = self.client.delete(
+            f'api/v1/contributions/{self.contribution.id}',
+            content_type='application/json', headers=self.admin_header)
+        self.assertTrue(mock.called)
```
Comment (georgreen) [inline test/api/views/test_notifications.py]: We had a discussion with @njirap and she adviced that we should assert that the mock was called with the arguments that were passed to it.

**Comment by Mnickii** [comment]
Please squash your commits

**Comment by georgreen** [comment]
Please rebase.

**Pull request closed without merging**